GeeksCoreLibrary/Modules/HealthChecks/Services/DatabaseHealthService.cs

```diff
@@ -0,0 +1,82 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+
+namespace GeeksCoreLibrary.Modules.HealthChecks.Services;
+public class DatabaseHealthService : IHealthCheck
+{
+    private readonly IDatabaseConnection databaseConnection;
+    private readonly HttpContext httpContext;
+    private readonly HealthChecksSettings healthChecksSettings;
+    
+    public DatabaseHealthService(IDatabaseConnection databaseConnection, IHttpContextAccessor httpContextAccessor, IOptions<HealthChecksSettings> healthChecksSettings)
+    {
+        this.databaseConnection = databaseConnection;
+        httpContext = httpContextAccessor.HttpContext;
+        this.healthChecksSettings = healthChecksSettings.Value;
+    }
+    
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        databaseConnection.ClearParameters();
+       
+        // A new query to check the max active connections from the DatabaseHealthCheck
+        var query =
+            $"SELECT COUNT(*) AS active_connections FROM information_schema.PROCESSLIST;";
+        var datatable = await databaseConnection.GetAsync(query);
+
+        if (datatable.Rows.Count == 0)
+        {
+            return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+        }
+         
+        var healthCheckConnections = healthChecksSettings.MaximumDatabaseConnections;
+        var healthCheckConnectionsTime = healthChecksSettings.MaximumConnectionsInTime;
+
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnections > 0)
+        {
+            // Retrieve the count of active connections.
+            var activeConnections = Convert.ToInt32(datatable.Rows[0]["active_connections"]); 
+            
+            // Check if the number of active connections exceeds the limit.
+            if (activeConnections > healthCheckConnections) 
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy(activeConnections +" databases connected, Too many database connections"));
+            }
+        }
+        
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnectionsTime > 0)
+        {    
+            // Query to check the max open connections in time from the DatabaseHealthCheck.
+            query =  $"SELECT ID, USER, HOST, TIME AS connection_time_in_sec FROM information_schema.PROCESSLIST";
+            datatable = await databaseConnection.GetAsync(query);
+
+            if (datatable.Rows.Count == 0)
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+            }
+            
+            // Retrieve the count of open connections in time.
+            var connectionTime = Convert.ToInt32(datatable.Rows[0]["connection_time_in_sec"]);
```

Comment:
Your query retrieves all open connections to the database without sorting or grouping them. Here you only get the first result and then check the connection time for that. This is not very useful, because now you check the connection time of a random connection, which doesn't tell us anything.

So you should either adjust your query so that it only retrieves the connection with the longest connection time, or add a `foreach` here and check each row separately.

Reply:
went for option 1, lets maybe give more info in v.1.1